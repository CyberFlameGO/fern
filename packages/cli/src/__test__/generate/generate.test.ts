import execa from "execa";
import { readdirSync } from "fs";
import { readFile, rm } from "fs/promises";
import path from "path";

const FIXTURES_DIR = path.join(__dirname, "fixtures");
const EXPECTED_FILES_TXT = "expectedFiles.txt";

describe("fern generate tests", () => {
    const fixtures = readdirSync(FIXTURES_DIR);

    for (const fixture of fixtures) {
        it(
            fixture,
            async () => {
                const fixturePath = path.join(FIXTURES_DIR, fixture);
                const outputPath = path.join(fixturePath, "generated");
                await rm(outputPath, { force: true, recursive: true });
                const cmd = execa("node", ["./cli", "generate", fixturePath], {
                    env: {
                        NODE_ENV: "development",
                    },
                });
                cmd.stdout?.pipe(process.stdout);
                cmd.stderr?.pipe(process.stderr);
                await cmd;

                const expectedFilesBuffer = await readFile(path.join(fixturePath, EXPECTED_FILES_TXT));
                const expectedFiles = expectedFilesBuffer
                    .toString()
                    .split("\n")
                    .map((s) => s.trim())
                    .filter((s) => s.length > 0);
                for (const expectedFile of expectedFiles) {
                    const fileContents = await readFile(path.join(outputPath, expectedFile));
                    expect(fileContents.toString()).toMatchSnapshot();
                }
            },
            10_000
        );
    }
});
