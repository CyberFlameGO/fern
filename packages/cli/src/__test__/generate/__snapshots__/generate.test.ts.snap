// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fern generate tests hathora 1`] = `
"import {
    defaultFetcher,
    Fetcher,
    joinPaths,
    Service,
} from \\"@fern-typescript/service-utils\\";
import * as model from \\"../../model\\";
import * as endpoints from \\"./endpoints\\";

export interface Client {
    deletePost(request: model.PostId): Promise<endpoints.deletePost.Response>;
}

export class Client implements Client {
    private baseUrl: string;
    private fetcher: Fetcher;
    private token: Service.Init[\\"token\\"];

    constructor(args: Service.Init) {
        this.fetcher = args.fetcher ?? defaultFetcher;
        this.baseUrl = joinPaths(args.serverUrl, args.serviceBaseUrl);
        this.token = args.token;
    }
}
"
`;

exports[`fern generate tests simple-model 1`] = `
"/**
 * A blog post
 */
export interface Post {
    id: string;
    title: string;
    content: string;
}
"
`;
